xbmc/addons/test/TestAddonInfoBuilder.cpp

```diff
@@ -156,6 +156,8 @@ TEST_F(TestAddonInfoBuilder, TestGenerate_DBEntry)
   EXPECT_TRUE(addon->HasType(ADDON_AUDIO));
   EXPECT_FALSE(addon->HasType(ADDON_GAME));
 
+  EXPECT_FALSE(addon->IsMainType(ADDON_AUDIO));
```

Comment:
why is this test failing? i just don't get it

Reply:
There becomes `addonType.Insert("provides", "video audio");` added, where marks it as `ADDON_VIDEO` and  `ADDON_AUDIO`.